pkg/apis/ceph.rook.io/v1/resources.go

```diff
@@ -60,11 +66,45 @@ func GetMonResources(p ResourceSpec) v1.ResourceRequirements {
 	return p[ResourcesKeyMon]
 }
 
-// GetOSDResources returns the placement for the OSDs
+// GetOSDResources returns the placement for the all OSDs
 func GetOSDResources(p ResourceSpec) v1.ResourceRequirements {
 	return p[ResourcesKeyOSD]
 }
 
+// GetOSDResourceKeyForDeviceClass returns key name for device class in resources spec
+func GetOSDResourceKeyForDeviceClass(deviceClass string) string {
+	return ResourcesKeyOSD + "-" + deviceClass
+}
+
+// GetOSDResourcesForDeviceClass returns the placement for the OSDs of specified device class
+func GetOSDResourcesForDeviceClass(p ResourceSpec, deviceKey string) v1.ResourceRequirements {
```

Comment:
It would be good to be able to handle a specific case, one can decide to create a DeviceClass named `foo` as well. However, it's still good to support known device classes as a start.